node/src/components/consensus/protocols/highway.rs

```diff
@@ -91,8 +91,10 @@ impl<I: NodeIdT, C: Context + 'static> HighwayProtocol<I, C> {
         let highway_config = &chainspec.genesis.highway_config;
 
         let total_weight = u128::from(validators.total_weight());
-        let ftt_percent = u128::from(highway_config.finality_threshold_percent);
-        let ftt = ((total_weight * ftt_percent / 100) as u64).into();
+        let ftt_fraction = highway_config.finality_threshold_fraction;
+        let ftt = ((total_weight * *ftt_fraction.numer() as u128 / *ftt_fraction.denom() as u128)
+            as u64)
```

Comment:
If `total_weight` is `u128` and `ftt_fraction` can be up to 1, can we safely cast `total_weight * ftt_fraction` to `u64`?

Reply:
Yes, because `total_weight` was cast from a `u64` above. We just need to make sure that the fraction is really <1.

Reply:
OK, but we don't do that yet as per your other comment.

Reply:
I've added some validation code now.